### PR TITLE
fix(link): match experimental spec

### LINKED ISSUE #1900: fix[Link component]: focus and active states
## fix[Link component]: focus and active states

(Right now there’s a 3px line underneath for both :focus and :active)

- [x] Change to a one pixel box for :focus (box is 18px in height; 1px weight; 2px offset on either side of type)

- [x] Make sure link token is $body-short-01

- [x] Only difference for :active is that it turns black: underline should stay the same as hover state

- [x] 32px of space between enabled and disabled link in component display window (has nothing to do with the component, this is just website display)

- [x] Restore default distance between underline and link (right now it looks like there’s an offset of 5px); the spec has Sketch default underline (looks like about 1px distance)

- [x] Need to fix Active state in Chrome and Firefox per issue https://github.com/IBM/carbon-components/issues/1909#issuecomment-467183947  (see spec)
<br>

Spec:

image

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1900

This PR incorporates feedback from the February 2019 audit of the link component

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]
👍

**Comment by emyarod** [comment]
moved back to WIP as we will be receiving a new spec for this component

**Comment by emyarod** [comment]
new spec (https://github.com/IBM/carbon-components/pull/1802#issuecomment-469351568)

> @emyarod 
> 
> Here's the updated spec as promised (linked on Airtable too):
> 
> https://ibm.box.com/s/wic90965yccngbmkt1hr6ycc8yw51g25
> 
> image
> 
> 
> Again we are going back to the original design in order to match digital now that they fixed their problem with the fat underline:
> 
> https://pages.github.ibm.com/webstandards/carbon-integration-poc/components/link.html
> 
>

**Review by jeanservaas — APPROVED** [review]
LGTM! thanks for your patience!

File: src/components/link/link.hbs [hunk 1]
```diff
@@ -5,10 +5,15 @@
   LICENSE file in the root directory of this source tree.
 -->
 
-<a href="#" class="{{@root.prefix}}--link">Link</a>
-<a class="{{@root.prefix}}--link">Placeholder link</a>
+<div style="margin-bottom: 32px">
```
Comment (asudoh) [inline src/components/link/link.hbs]: Is this demo-only element/style? If so, can we add styles to `demo.scss` with `.component-example__live--rendered.link` selector...?

**Review by asudoh — COMMENTED** [review]
LGTM basically, just one thing - Thanks @emyarod!

**Comment by tw15egan** [comment]
Looks good once Akira's changes are made

**Comment by asudoh** [comment]
@emyarod Do you think markup change is still needed given you got control over demo style? Or does the markup change include non-demo ones...?

**Comment by emyarod** [comment]
do you mean the `<div>` wrappers @asudoh? I can avoid that by changing the `display: inline` rule of the anchors in the examples if we would prefer to not wrap them with divs

**Comment by asudoh** [comment]
Great thanks @emyarod - Yeah didn't want to change the markup unless consumers need it - Though I suspect not many people are using it, but we are shipping the markup as we got a request in the past.

**Comment by emyarod** [comment]
got it, I refactored the demo to avoid modifying the component styles and the example markup @asudoh

File: demo/scss/demo.scss [hunk 2]
```diff
@@ -68,6 +68,11 @@ REASON: #{$reason}';
     .bx--interior-left-nav.bx--interior-left-nav--collapsed {
       width: 3rem;
     }
+
+    &.link div {
+      display: grid;
```
Comment (asudoh) [inline demo/scss/demo.scss]: Does this work well for IE11...? Probably a flex box?

File: src/components/link/link.hbs [hunk 3]
```diff
@@ -6,9 +6,8 @@
 -->
 
 <a href="#" class="{{@root.prefix}}--link">Link</a>
-<a class="{{@root.prefix}}--link">Placeholder link</a>
-<p class="{{@root.prefix}}--link--disabled">Disabled Link</p>
-<br>
 <a href="#" class="{{@root.prefix}}--link">Lorem ipsum dolor sit amet consectetur adipisicing elit. Corrupti explicabo
   nobis consequatur quod qui quam laboriosam placeat repudiandae. Ducimus ea aut omnis asperiores mollitia
   necessitatibus architecto temporibus provident quo? Repellat!</a>
+<a class="{{@root.prefix}}--link">Placeholder link</a>
```
Comment (asudoh) [inline src/components/link/link.hbs]: Do we have to move around the order...?

**Review by asudoh — COMMENTED** [review]
Thank you for your quick update @emyarod! Getting there, just a couple of remaining comments.

File: src/components/link/link.hbs (hunk 3 above)
Comment (emyarod) [reply src/components/link/link.hbs]: I thought it made sense to group the valid/invalid states together, but I reverted the change

File: demo/scss/demo.scss (hunk 2 above)
Comment (emyarod) [reply demo/scss/demo.scss]: I replaced the grid with flexbox

**Pull request merged**